File: src/fix_braces.js
#!/usr/bin/env node
/**
 * Script to fix missing closing braces in index.tsx
 * Adds two missing closing braces after line 2678
 * 
 * Usage: node fix_braces.js
 */

const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'server', 'index.tsx');

console.log(`Reading file: ${filePath}`);

// Read the file
const content = fs.readFileSync(filePath, 'utf8');
const lines = content.split('\n');

console.log(`Total lines: ${lines.length}`);
console.log(`\nLine 2678: "${lines[2677]}"`);
console.log(`Line 2679: "${lines[2678]}"`);

// Insert the missing braces after line 2678 (index 2677 in 0-based array)
// We need to add after line 2678:
//   "                }"  (16 spaces - closes if (allDealItems))
//   "            }"  (12 spaces - closes if (!error && wonDeals))

// Insert in correct order
lines.splice(2678, 0, "                }", "            }");

console.log(`\n✅ Inserted 2 lines after line 2678`);
console.log(`New line 2679: "${lines[2678]}"`);
console.log(`New line 2680: "${lines[2679]}"`);
console.log(`New line 2681: "${lines[2680]}"`);

// Write back
fs.writeFileSync(filePath, lines.join('\n'), 'utf8');

console.log(`\n✅ File saved successfully!`);
console.log(`\nFixed braces:`);
console.log(`  - Line 2679: '                }' (16 spaces - closes if (allDealItems))`);
console.log(`  - Line 2680: '            }' (12 spaces - closes if (!error && wonDeals))`);

File: src/fix_braces.py
#!/usr/bin/env python3
"""
Script to fix missing closing braces in index.tsx
Adds two missing closing braces after line 2678
"""

file_path = 'src/server/index.tsx'

# Read the file
with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Insert the missing braces after line 2678 (index 2677 in 0-based array)
# Line 2678 contains: "             }"
# After it, we need to add:
#   Line 2679 (new): "                }\n"  (16 spaces - closes if (allDealItems))
#   Line 2680 (new): "            }\n"  (12 spaces - closes if (!error && wonDeals))

# Insert in reverse order to maintain correct line numbers
lines.insert(2678, "            }\n")  # 12 spaces
lines.insert(2678, "                }\n")  # 16 spaces

# Write back
with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(lines)

print("✅ Fixed! Added missing closing braces after line 2678.")
print("Lines inserted:")
print("  - Line 2679: '                }' (16 spaces - closes if (allDealItems))")
print("  - Line 2680: '            }' (12 spaces - closes if (!error && wonDeals))")
